configmap/parse.go

```diff
@@ -144,6 +146,47 @@ func AsQuantity(key string, target **resource.Quantity) ParseFunc {
 	}
 }
 
+// AsOptionalNamespacedName parses the value at key as a types.NamespacedName into the target, if it exists
+// The namespace and name are both required and expected to be valid DNS labels
+func AsOptionalNamespacedName(key string, target **types.NamespacedName) ParseFunc {
```

Comment:
Can't we just use empty NN as absent?

Reply:
empty NN isn't technically valid but maybe 🤷. If we're going to have to check `target != types.NamespacedName{}` seems probably clearer to say `target != nil`?

Reply:
nil check will be more succinct